examples/ft-refund/solidity-contract/src/FtRefund.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: CC-BY-1.0
+pragma solidity ^0.8.17;
+
+// See `docs/NearFromAurora.md` for more information on what is going on in this example.
+
+import "openzeppelin-contracts/access/AccessControl.sol";
+import "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/utils/Strings.sol";
+import "./IEvmErc20.sol";
+import {AuroraSdk, Codec, NEAR, PromiseCreateArgs, PromiseResult, PromiseResultStatus, PromiseWithCallback} from "aurora-sdk/AuroraSdk.sol";
+
+uint64 constant FT_TRANSFER_CALL_NEAR_GAS = 35_000_000_000_000;
+
+uint64 constant CALLBACK_NEAR_GAS = 130_000_000_000_000;
+
+uint64 constant REFUND_NEAR_GAS = 35_000_000_000_000;
+
+uint64 constant APPROVE_NEAR_GAS = 20_000_000_000_000;
+
+contract FtRefund is AccessControl {
+    using AuroraSdk for NEAR;
+    using AuroraSdk for PromiseCreateArgs;
+    using AuroraSdk for PromiseWithCallback;
+    using Codec for bytes;
+
+    bytes32 public constant CALLBACK_ROLE = keccak256("CALLBACK_ROLE");
+    bytes16 private constant _SYMBOLS = "0123456789abcdef";
+
+    IERC20 public wNEAR;
+    string public nearAccountId;
+    NEAR public near;
+
+    constructor(string memory _nearAccountId, IERC20 _wNEAR) {
+        nearAccountId = _nearAccountId;
+        near = AuroraSdk.initNear(_wNEAR);
+        wNEAR = _wNEAR;
+        _grantRole(
+            CALLBACK_ROLE,
+            AuroraSdk.nearRepresentitiveImplicitAddress(address(this))
+        );
+    }
+
+    function approveWNEAR() public {
```

Comment:
If I correctly understood the meaning of this function -- `approve` will be called by CALLBACK_ROLE and we will allowe this contract to spend the CALLBACK_ROLE's wNEAR. This should be explained in the comments to this function. And who provided wNEAR to the CALLBACK_ROLE?

Reply:
This is the tricky part of this example (and a sort of pitfall in the XCC design). The `CALLBACK_ROLE` is logically the same as this contract because it is the implicit address of this contract's XCC account (and therefore all transactions from `CALLBACK_ROLE` actually originated from this contract). So this function is essentially just approving itself to spend its own tokens. The reason its needed is because the addresses are different on the level of the EVM even though they are logically the same entity as I have described above. I tried to explain this in the README file and comments in the integration tests. But I could add a comment here too.

I think it is the maintainer of this contract who is responsible for funding WNEAR to `CALLBACK_ROLE`. It only requires 1 yoctoNEAR per call, so if a user were to try to maliciously drain the WNEAR by intentionally making failed calls they would need to invest a lot of ETH or NEAR in gas fees. If the maintainer initially funded it with 1 WNEAR then 10^24 refunds would be needed to drain it and that is infeasible.

Reply:
Added a comment in https://github.com/aurora-is-near/aurora-contracts-sdk/pull/13/commits/6180a22a54214c2cc6f726666f5207f9637f5c08